### PR TITLE
Add Supervisor Internal Field Functions

### PR DESCRIPTION
**Pull request by DavidMansolino**
**Related Issues**
This pull-request fixes issue #1310

### CODE REVIEW

**Comment by DavidMansolino** [comment]
:heavy_check_mark: travis_distrib

File: src/webots/nodes/utils/WbSupervisorUtilities.cpp [hunk 1]
```diff
@@ -863,24 +866,29 @@ void WbSupervisorUtilities::handleMessage(QDataStream &stream) {
...
       mFoundFieldType = 0;
       mFoundFieldCount = -1;
+      mFoundFieldVisible = false;
 
       WbNode *const node = WbNode::findNode(id);
       if (node) {
-        // qDebug() << "Node=" << node->fullName() << " uniqueId=" << id << " fieldName=" << name;
-        id = node->findFieldId(name);
+        // qDebug() << "Node=" << node->fullName() << " uniqueId=" << id << " fieldName=" << name " allowSearchInProto=" <<
+        // allowSearchInProto == 1;
```
Comment (omichel) [inline src/webots/nodes/utils/WbSupervisorUtilities.cpp]: Shall we remove this line?

**Comment by DavidMansolino** [comment]
:heavy_check_mark:: travis_unit + successfully tested in  cyberbotics/webots_ros2#50

File: src/webots/vrml/WbNode.hpp [hunk 2]
```diff
@@ -145,6 +145,9 @@ class WbNode : public QObject {
   int useCount() const { return mUseNodes.size(); }
   const QList<WbNode *> &useNodes() const { return mUseNodes; }
   virtual void defHasChanged() {}
+  // Look for a node with this DEF names (this node or any of its children nodes)
```
Comment (omichel) [inline src/webots/vrml/WbNode.hpp]:
```suggestion
  // Look for a node with this DEF name (this node or any of its children nodes)
```

File: tests/api/controllers/supervisor_field/supervisor_field.c [hunk 3]
```diff
@@ -70,6 +70,25 @@ int main(int argc, char **argv) {
   d = wb_supervisor_field_get_sf_float(field);
   ts_assert_double_equal(d, 0.73, "Proto \"camera_fieldOfView\" SFFloat field should have value 0.73 not %f", d);
 
+  // internal SFNode
+  internal_sf_node_field = wb_supervisor_node_get_proto_field(proto, "physics");
+  ts_assert_pointer_not_null(internal_sf_node_field, "wb_supervisor_node_get_proto_field should return internal fields.");
+  internal_physics = wb_supervisor_field_get_sf_node(internal_sf_node_field);
+  ts_assert_pointer_not_null(internal_physics, "wb_supervisor_field_get_sf_node should return internal field value.");
```
Comment (omichel) [inline tests/api/controllers/supervisor_field/supervisor_field.c]:
```suggestion
  ts_assert_pointer_not_null(internal_physics, "wb_supervisor_field_get_sf_node should return an internal node.");
```

File: tests/api/controllers/supervisor_field/supervisor_field.c [hunk 4]
```diff
@@ -70,6 +70,25 @@ int main(int argc, char **argv) {
   d = wb_supervisor_field_get_sf_float(field);
   ts_assert_double_equal(d, 0.73, "Proto \"camera_fieldOfView\" SFFloat field should have value 0.73 not %f", d);
 
+  // internal SFNode
+  internal_sf_node_field = wb_supervisor_node_get_proto_field(proto, "physics");
+  ts_assert_pointer_not_null(internal_sf_node_field, "wb_supervisor_node_get_proto_field should return internal fields.");
```
Comment (omichel) [inline tests/api/controllers/supervisor_field/supervisor_field.c]:
```suggestion
  ts_assert_pointer_not_null(internal_sf_node_field, "wb_supervisor_node_get_proto_field should return an internal field.");
```

File: tests/api/controllers/supervisor_field/supervisor_field.c [hunk 5]
```diff
@@ -70,6 +70,25 @@ int main(int argc, char **argv) {
...
+  internal_physics = wb_supervisor_field_get_sf_node(internal_sf_node_field);
+  ts_assert_pointer_not_null(internal_physics, "wb_supervisor_field_get_sf_node should return internal field value.");
+  internal_field = wb_supervisor_node_get_proto_field(internal_physics, "density");
+  ts_assert_pointer_not_null(internal_field, "wb_supervisor_node_get_proto_field should return internal fields.");
```
Comment (omichel) [inline tests/api/controllers/supervisor_field/supervisor_field.c]:
```suggestion
  ts_assert_pointer_not_null(internal_field, "wb_supervisor_node_get_proto_field should return an internal field.");
```

File: tests/api/controllers/supervisor_field/supervisor_field.c [hunk 6]
```diff
@@ -70,6 +70,25 @@ int main(int argc, char **argv) {
...
+  d = wb_supervisor_field_get_sf_float(internal_field);
+  ts_assert_double_equal(d, -1.0, "Returned value should be -1.0 and not %f", d);
+
+  // internal SFFloat value
+  internal_field = wb_supervisor_node_get_field(proto, "cpuConsumption");
+  ts_assert_pointer_null(internal_field, "wb_supervisor_node_get_field should not return internal fields.");
+  internal_field = wb_supervisor_node_get_proto_field(proto, "cpuConsumption");
+  ts_assert_pointer_not_null(internal_field, "wb_supervisor_node_get_proto_field should return internal fields.");
```
Comment (omichel) [inline tests/api/controllers/supervisor_field/supervisor_field.c]:
```suggestion
  ts_assert_pointer_not_null(internal_field, "wb_supervisor_node_get_proto_field should return an internal field.");
```

**Pull request merged**